rclrs/src/node/graph.rs

```diff
@@ -0,0 +1,635 @@
+                &mut count,
+            )
+            .ok()?
+        };
+        Ok(count)
+    }
+
+    /// Count the number of subscriptions for a given topic.
+    pub fn count_subscriptions(&self, topic: &str) -> Result<usize, RclrsError> {
+        let topic_name = CString::new(topic).map_err(|err| RclrsError::StringContainsNul {
+            s: topic.to_string(),
+            err,
+        })?;
+        let mut count: usize = 0;
+
+        // SAFETY: The topic_name string was correctly allocated previously
+        unsafe {
+            rcl_count_subscribers(
+                &*self.rcl_node_mtx.lock().unwrap(),
+                topic_name.as_ptr(),
+                &mut count,
+            )
+            .ok()?
+        };
+        Ok(count)
+    }
+
+    /// Get topic publisher info.
+    pub fn get_publishers_info_by_topic(
+        &self,
+        topic: &str,
+    ) -> Result<Vec<TopicEndpointInfo>, RclrsError> {
+        self.get_publisher_subscriber_info_by_topic(topic, rcl_get_publishers_info_by_topic)
+    }
+
+    /// Get topic subscriptions info.
+    pub fn get_subscriptions_info_by_topic(
+        &self,
+        topic: &str,
+    ) -> Result<Vec<TopicEndpointInfo>, RclrsError> {
+        self.get_publisher_subscriber_info_by_topic(topic, rcl_get_subscriptions_info_by_topic)
+    }
+
+    /// Get an rcl names_and_types function, with a "no_demangle" argument.
+    fn get_names_and_types_by_node_with_demangle_arg(
+        &self,
+        node: &str,
+        namespace: &str,
+        getter: unsafe extern "C" fn(
+            *const rcl_node_t,
+            *mut rcl_allocator_t,
+            bool,
+            *const ::std::os::raw::c_char,
+            *const ::std::os::raw::c_char,
+            *mut rcl_names_and_types_t,
+        ) -> rcl_ret_t,
+    ) -> Result<TopicNamesAndTypes, RclrsError> {
+        // SAFETY: Getting a zero-initialized value is always safe
+        let mut rcl_names_and_types = unsafe { rmw_get_zero_initialized_names_and_types() };
+
+        let node_name = CString::new(node).map_err(|err| RclrsError::StringContainsNul {
+            s: node.to_string(),
+            err,
+        })?;
+        let node_namespace =
+            CString::new(namespace).map_err(|err| RclrsError::StringContainsNul {
+                s: namespace.to_string(),
+                err,
+            })?;
+
+        // SAFETY: node_name and node_namespace have been zero-initialized.
+        unsafe {
+            getter(
+                &*self.rcl_node_mtx.lock().unwrap() as *const _,
+                &mut rcutils_get_default_allocator(),
+                false,
+                node_name.as_ptr(),
+                node_namespace.as_ptr(),
+                &mut rcl_names_and_types,
+            )
+        };
+
+        Ok(convert_names_and_types(rcl_names_and_types))
+    }
+
+    /// Get an rcl names_and_types function, without a "no_demangle" argument.
+    fn get_names_and_types_by_node_without_demangle_arg(
```

Comment:
Do you think it would be worth unifying the version with a demangle arg with this function, à la https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b173ae6ee24b818a94bac23c6d9142b9?